cli/src/pcluster/validators/s3_validators.py

```diff
@@ -64,3 +66,62 @@ def _validate_s3_uri(self, s3_url):
                 ),
                 FailureLevel.ERROR,
             )
+
+
+class S3BucketUriValidator(Validator):
+    """S3 Bucket Url Validator."""
+
+    def _validate(self, url):
+
+        if urlparse(url).scheme == "s3":
+            try:
+                bucket = get_bucket_name_from_s3_url(url)
+                S3Client().head_bucket(bucket_name=bucket)
+            except ClientError as client_error:
+                message = _process_generic_s3_bucket_error(client_error, bucket)
+                self._add_failure(message, FailureLevel.ERROR)
+        else:
+            self._add_failure(f"The value '{url}' is not a valid S3 URI.", FailureLevel.ERROR)
+
+
+class S3BucketValidator(Validator):
+    """S3 Bucket Validator."""
+
+    def _validate(self, bucket):
+
+        s3_client = S3Client()
+        try:
+            s3_client.head_bucket(bucket_name=bucket)
+            # Check versioning is enabled on the bucket
+            response = s3_client.get_bucket_versioning(bucket_name=bucket)
+            if response.get("Status") != "Enabled":
+                self._add_failure(
+                    "The S3 bucket {0} specified cannot be used by cluster "
+                    "because versioning setting is: {1}, not 'Enabled'. Please enable bucket versioning.".format(
+                        bucket, response.get("Status")
+                    ),
+                    FailureLevel.ERROR,
+                )
+
+        except ClientError as client_error:
+            message = _process_generic_s3_bucket_error(client_error, bucket)
+            self._add_failure(message, FailureLevel.ERROR)
+        except ParamValidationError as validation_error:
+            self._add_failure(
+                "Error validating parameter '{0}'. Failed with exception: {1}".format(bucket, str(validation_error)),
+                FailureLevel.ERROR,
+            )
+
+
+def _process_generic_s3_bucket_error(client_error, bucket_name):
```

Comment:
You're using S3Client, it means that the ClientError will be captured by the boto3 layer and you'll have instead a AWSClientError.

You should move this logic in the boto3 layer.

Reply:
Done